jetty-core/jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/ResponseHeaderRegexRule.java

```diff
@@ -0,0 +1,138 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.rewrite.handler;
+
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.StringUtil;
+
+/**
+ * Sends the response code whenever the rule finds a match.
+ */
+public class ResponseHeaderRegexRule extends Rule
+{
+    private String _headerName;
+    private Pattern _headerRegex;
+    private int _code;
+    private String _message;
+
+    public ResponseHeaderRegexRule()
+    {
+    }
+
+    @Override
+    public boolean isTerminating()
+    {
+        return true;
+    }
+
+    public String getHeaderName()
+    {
+        return _headerName;
+    }
+
+    public void setHeaderName(String header)
+    {
+        _headerName = header;
+    }
+
+    public String getHeaderRegex()
+    {
+        return _headerRegex == null ? null : _headerRegex.pattern();
+    }
+
+    public void setHeaderRegex(String headerRegex)
+    {
+        _headerRegex = headerRegex == null ? null : Pattern.compile(headerRegex);
+    }
+
+    public int getCode()
+    {
+        return _code;
+    }
+
+    /**
+     * Set the response code.
+     * @param code the response code
+     */
+    public void setCode(int code)
+    {
+        _code = code;
+    }
+
+    public String getMessage()
+    {
+        return _message;
+    }
+
+    /**
+     * <p>Sets the message for the response body (if the response code may have a body).</p>
+     *
+     * @param message the response message
+     */
+    public void setMessage(String message)
+    {
+        _message = message;
+    }
+
+    @Override
+    public Handler matchAndApply(Handler input) throws IOException
+    {
+        String value = input.getHeaders().get(getHeaderName());
+        if (value == null)
+            return null;
+        if (_headerRegex == null)
+            return apply(input, value);
+        Matcher matcher = _headerRegex.matcher(value);
+        if (matcher.matches())
+            return apply(input, value);
+        return null;
+    }
+
+    public Handler apply(Handler input, String value) throws IOException
+    {
+        if (getCode() < HttpStatus.CONTINUE_100)
+            return null;
```

Comment:
Move this check to `setCode()` (to be renamed to `setStatus()`) and throw `IllegalArgumentException`.

Reply:
Ok, will do. This construct was based on the existing ResponsePatternRule.java code.